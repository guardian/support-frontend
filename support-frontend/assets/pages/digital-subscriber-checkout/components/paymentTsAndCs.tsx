import { css } from '@emotion/react';
import { neutral, textSans12 } from '@guardian/source/foundations';
import { StripeDisclaimer } from 'components/stripe/stripeDisclaimer';
import { privacyLink } from 'helpers/legal';

const marginTop = css`
	margin-top: 4px;
`;

const container = css`
	${textSans12};
	color: ${neutral[20]};

	& a {
		color: ${neutral[20]};
	}
`;

export function PaymentTsAndCs(): JSX.Element {
	return (
		<div css={container}>
			Payment taken after the first 14 day free trial. At the end of the free
			trial period your subscription will auto-renew, and you will be charged,
			each month at the full price of £14.99 per month or £149 per year unless
			you cancel. You can cancel at any time before your next renewal date.
			Cancellation will take effect at the end of your current subscription
			month. To cancel, go to{' '}
			<a href="http://manage.theguardian.com/">Manage My Account</a> or see our{' '}
			<a href="https://www.theguardian.com/info/2014/aug/06/guardian-observer-digital-subscriptions-terms-conditions">
				Terms
			</a>
			.
			<div css={marginTop}>
				By proceeding, you are agreeing to our{' '}
				<a href="https://www.theguardian.com/info/2014/aug/06/guardian-observer-digital-subscriptions-terms-conditions">
					Terms and Conditions
				</a>
				.{' '}
				<p css={marginTop}>
					To find out what personal data we collect and how we use it, please
					visit our <a href={privacyLink}>Privacy Policy</a>.
				</p>
				<p css={marginTop}>
					<StripeDisclaimer />
				</p>
			</div>
		</div>
	);
}
