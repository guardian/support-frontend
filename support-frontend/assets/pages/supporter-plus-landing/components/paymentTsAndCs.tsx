import { css } from '@emotion/react';
import { neutral, textSans } from '@guardian/source-foundations';
import type {
	ContributionType,
	RegularContributionType,
} from 'helpers/contributions';
import { formatAmount } from 'helpers/forms/checkouts';
import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
import type { IsoCurrency } from 'helpers/internationalisation/currency';
import {
	currencies,
	spokenCurrencies,
} from 'helpers/internationalisation/currency';
import { contributionsTermsLinks, privacyLink } from 'helpers/legal';
import { supporterPlusLegal } from 'helpers/legalCopy';
import type { Promotion } from 'helpers/productPrice/promotions';
import { sendTrackingEventsOnClick } from 'helpers/productPrice/subscriptions';
import { manageSubsUrl } from 'helpers/urls/externalLinks';
import {
	getDateWithOrdinal,
	getLongMonth,
} from 'helpers/utilities/dateFormatting';
import type { FinePrintTheme } from './finePrint';
import { FinePrint } from './finePrint';

const marginTop = css`
	margin-top: 4px;
`;

const container = css`
	${textSans.xxsmall()};
	color: ${neutral[20]};

	& a {
		color: ${neutral[20]};
	}
`;

interface PaymentTsAndCsProps {
	mobileTheme?: FinePrintTheme;
	contributionType: ContributionType;
	countryGroupId: CountryGroupId;
	currency: IsoCurrency;
	amount: number;
	amountIsAboveThreshold: boolean;
	productNameAboveThreshold: string;
	promotion?: Promotion;
}

const manageMyAccount = (
	<a
		href={manageSubsUrl}
		onClick={sendTrackingEventsOnClick({
			id: 'checkout_my_account',
			product: 'PremiumTier',
			componentType: 'ACQUISITIONS_BUTTON',
		})}
	>
		Manage My Account
	</a>
);

const termsSupporterPlus = (linkText: string) => (
	<a href="https://www.theguardian.com/info/2022/oct/28/the-guardian-supporter-plus-terms-and-conditions">
		{linkText}
	</a>
);

function TsAndCsRenewal({
	contributionType,
}: {
	contributionType: ContributionType;
}): JSX.Element {
	const today = new Date();
	if (contributionType === 'ANNUAL') {
		return (
			<>
				on the {getDateWithOrdinal(today)} day of {getLongMonth(today)} every{' '}
				year
			</>
		);
	}
	return <>on the {getDateWithOrdinal(today)} day of every month</>;
}

function TsAndCsFooterLinks({
	countryGroupId,
	amountIsAboveThreshold,
}: {
	countryGroupId: CountryGroupId;
	amountIsAboveThreshold: boolean;
}) {
	const privacy = <a href={privacyLink}>Privacy Policy</a>;

	const termsContributions = (
		<a href={contributionsTermsLinks[countryGroupId]}>Terms and Conditions</a>
	);

	const terms = amountIsAboveThreshold
		? termsSupporterPlus('Terms and Conditions')
		: termsContributions;

	return (
		<div css={marginTop}>
			By proceeding, you are agreeing to our {terms}.{' '}
			<p css={marginTop}>
				To find out what personal data we collect and how we use it, please
				visit our {privacy}.
			</p>
		</div>
	);
}

export function PaymentTsAndCs({
	mobileTheme = 'dark',
	contributionType,
	countryGroupId,
	currency,
	amount,
	amountIsAboveThreshold,
	productNameAboveThreshold,
	promotion,
}: PaymentTsAndCsProps): JSX.Element {
	const amountCopy = isNaN(amount)
		? null
		: ` of ${formatAmount(
				currencies[currency],
				spokenCurrencies[currency],
				amount,
				false,
		  )}`;

	const frequencySingular = (contributionType: ContributionType) =>
		contributionType === 'MONTHLY' ? 'month' : 'year';

	const frequencyPlural = (contributionType: ContributionType) =>
		contributionType === 'MONTHLY' ? 'monthly' : 'annual';

	const copyAboveThreshold = (
		contributionType: RegularContributionType,
		productNameAboveThreshold: string,
		promotion?: Promotion,
	) => {
		return (
			<>
				<div>
					If you pay at least{' '}
					{supporterPlusLegal(
						countryGroupId,
						contributionType,
						' per ',
						promotion,
					)}
					, you will receive the {productNameAboveThreshold} benefits on a
					subscription basis. If you increase your payments per{' '}
					{frequencySingular(contributionType)}, these additional amounts will
					be separate {frequencyPlural(contributionType)} voluntary financial
					contributions to the Guardian. The {productNameAboveThreshold}{' '}
					subscription and any contributions will auto-renew each{' '}
					{frequencySingular(contributionType)}. You will be charged the
					subscription and contribution amounts using your chosen payment method
					at each renewal unless you cancel. You can cancel your subscription or
					change your contributions at any time before your next renewal date.
					If you cancel within 14 days of taking out a{' '}
					{productNameAboveThreshold} subscription, youâ€™ll receive a full refund
					(including of any contributions) and your subscription and any
					contribution will stop immediately. Cancellation of your subscription
					(which will also cancel any contribution) or cancellation of your
					contribution made after 14 days will take effect at the end of your
					current {frequencyPlural(contributionType)} payment period. To cancel,
					go to {manageMyAccount} or see our {termsSupporterPlus('Terms')}.
				</div>
				<TsAndCsFooterLinks
					countryGroupId={countryGroupId}
					amountIsAboveThreshold={amountIsAboveThreshold}
				/>
			</>
		);
	};

	if (contributionType === 'ONE_OFF') {
		return (
			<div css={container}>
				<FinePrint mobileTheme={mobileTheme}>
					<TsAndCsFooterLinks
						countryGroupId={countryGroupId}
						amountIsAboveThreshold={amountIsAboveThreshold}
					/>
				</FinePrint>
			</div>
		);
	}

	const copyBelowThreshold = (contributionType: ContributionType) => {
		return (
			<>
				<div>
					We will attempt to take payment{amountCopy},{' '}
					<TsAndCsRenewal contributionType={contributionType} />, from now until
					you cancel your payment. Payments may take up to 6 days to be recorded
					in your bank account. You can change how much you give or cancel your
					payment at any time.
				</div>
				<TsAndCsFooterLinks
					countryGroupId={countryGroupId}
					amountIsAboveThreshold={amountIsAboveThreshold}
				/>
			</>
		);
	};

	return (
		<div css={container}>
			<FinePrint mobileTheme={mobileTheme}>
				{amountIsAboveThreshold
					? copyAboveThreshold(
							contributionType,
							productNameAboveThreshold,
							promotion,
					  )
					: copyBelowThreshold(contributionType)}
			</FinePrint>
		</div>
	);
}
