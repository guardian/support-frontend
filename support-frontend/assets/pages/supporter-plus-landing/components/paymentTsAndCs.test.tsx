import { render } from '@testing-library/react';
import type { ContributionType } from 'helpers/contributions';
import type { CountryGroupId } from 'helpers/internationalisation/countryGroup';
import type { ActiveProductKey } from 'helpers/productCatalog';
import type { Promotion } from 'helpers/productPrice/promotions';
import { PaymentTsAndCs } from './paymentTsAndCs';

describe('Payment Ts&Cs Snapshot comparison', () => {
	const promotionTierThreUnitedStatesMonthly: Promotion = {
		name: '$8 off for 12 months',
		description: 'Tier Three United States Monthly',
		promoCode: 'TIER_THREE_USA_MONTHLY',
		numberOfDiscountedPeriods: 12,
		discountedPrice: 37,
	};
	const paymentProductKeys = [
		['GuardianAdLite', 'ANNUAL', 'GBPCountries', 0],
		['DigitalSubscription', 'MONTHLY', 'GBPCountries', 0],
		['OneTimeContribution', 'MONTHLY', 'UnitedStates', 0],
		['Contribution', 'ANNUAL', 'AUDCountries', 0],
		['SupporterPlus', 'MONTHLY', 'GBPCountries', 12],
		['TierThree', 'MONTHLY', 'UnitedStates', 45],
		['HomeDelivery', 'MONTHLY', 'GBPCountries', 0],
		['NationalDelivery', 'MONTHLY', 'GBPCountries', 0],
		['SubscriptionCard', 'MONTHLY', 'GBPCountries', 0],
	];
	it.each(paymentProductKeys)(
		`paymentTs&Cs render product %s for region %s correctly`,
		(paymentProductKey, contributionType, countryGroupId, amount) => {
			const promo: Promotion | undefined =
				paymentProductKey === 'TierThree' &&
				contributionType === 'MONTHLY' &&
				countryGroupId === 'UnitedStates'
					? promotionTierThreUnitedStatesMonthly
					: undefined;
			const { container } = render(
				<PaymentTsAndCs
					contributionType={contributionType as ContributionType}
					countryGroupId={countryGroupId as CountryGroupId}
					productKey={paymentProductKey as ActiveProductKey}
					thresholdAmount={amount as number}
					promotion={promo}
				/>,
			);
			expect(container).toMatchSnapshot();
		},
	);
});
