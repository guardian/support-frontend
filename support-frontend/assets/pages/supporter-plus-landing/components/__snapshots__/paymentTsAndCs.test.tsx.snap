// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product Contribution for region Annual correctly 1`] = `"By proceeding, you are agreeing to our Terms and Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product DigitalSubscription for region Monthly correctly 1`] = `"Payment taken after the first 14 day free trial. At the end of the free trial period your subscription will auto-renew, and you will be charged, each month at the full price of £14.99 per month or £149 per year unless you cancel. You can cancel at any time before your next renewal date. Cancellation will take effect at the end of your current subscription month. To cancel, go to Manage My Account or see our Terms.By proceeding, you are agreeing to our Terms and Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product GuardianAdLite for region Annual correctly 1`] = `"Your Guardian Ad-Lite subscription will auto-renew each year unless cancelled. Your first payment will be taken on day 15 after signing up but you will start to receive your Guardian Ad-Lite benefits when you sign up. Unless you cancel, subsequent monthly payments will be taken on this date using your chosen payment method. You can cancel your subscription at any time before your next renewal date. If you cancel your Guardian Ad-Lite subscription within 14 days of signing up, your subscription will stop immediately and we will not take the first payment from you. Cancellation of your subscription after 14 days will take effect at the end of your current monthly payment period. To cancel, go to Manage My Account or see our Guardian Ad-Lite Terms.By proceeding, you are agreeing to the Guardian Ad-Lite Terms. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product GuardianWeeklyDomestic for region Monthly correctly 1`] = `"By proceeding, you are agreeing to our Terms & Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product GuardianWeeklyRestOfWorld for region Annual correctly 1`] = `"By proceeding, you are agreeing to our Terms & Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product HomeDelivery for region EverydayPlus correctly 1`] = `"Your first payment will be taken on Sunday, September 21, 2025 when your first newspaper is delivered. You can cancel your subscription at any time before your next renewal date. Cancellation will take effect at the end of your current payment period. To cancel, use the contact details listed on our Help Centre. We will share your contact and subscription details with our fulfilment partners.By proceeding, you are agreeing to our Terms & Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product NationalDelivery for region EverydayPlus correctly 1`] = `"Your first payment will be taken on Sunday, September 21, 2025 when your first newspaper is delivered. You can cancel your subscription at any time before your next renewal date. Cancellation will take effect at the end of your current payment period. To cancel, use the contact details listed on our Help Centre. We will share your contact and subscription details with our fulfilment partners.By proceeding, you are agreeing to our Terms & Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product SubscriptionCard for region EverydayPlus correctly 1`] = `"Your first payment will be taken on Sunday, September 21, 2025. You can cancel your subscription at any time before your next renewal date. Cancellation will take effect at the end of your current payment period. To cancel, use the contact details listed on our Help Centre. We will share your contact and subscription details with our fulfilment partners to provide you with your subscription card.By proceeding, you are agreeing to our Terms & Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product SupporterPlus for region Monthly correctly 1`] = `"If you pay at least £12 per month, you will receive the All-access digital benefits on a subscription basis. If you increase your payments per month, these additional amounts will be separate monthly voluntary financial contributions to the Guardian. The All-access digital subscription and any contributions will auto-renew each month. You will be charged the subscription and contribution amounts using your chosen payment method at each renewal unless you cancel. You can cancel your subscription or change your contributions at any time before your next renewal date. If you cancel within 14 days of taking out a All-access digital subscription, you’ll receive a full refund (including of any contributions) and your subscription and any contribution will stop immediately. Cancellation of your subscription (which will also cancel any contribution) or cancellation of your contribution made after 14 days will take effect at the end of your current monthly payment period. To cancel, go to Manage My Account or see our Terms.By proceeding, you are agreeing to our Terms and Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product SupporterPlus for region OneYearStudent correctly 1`] = `"You may cancel your All-access digital subscription within 14 days of taking out the subscription. If you do, you'll receive a full refund and your subscription will stop immediately. To cancel, go to Manage My Account or see our Terms. This subscription does not auto-renew.By proceeding, you are agreeing to our Terms and Conditions. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;

exports[`Payment Ts&Cs Snapshot comparison paymentTs&Cs render product TierThree for region RestOfWorldMonthly correctly 1`] = `"By signing up, you are taking out a Digital + print subscription. Your Digital + print subscription will auto-renew each month unless cancelled. Your first payment will be taken on the publication date of your first Guardian Weekly magazine (as shown in the checkout) but you will start to receive your digital benefits when you sign up. Unless you cancel, subsequent monthly payments will be taken on this date using your chosen payment method. You can cancel your Digital + print subscription at any time before your next renewal date. If you cancel your Digital + print subscription within 14 days of signing up, your subscription will stop immediately and we will not take the first payment from you. Cancellation of your subscription after 14 days will take effect at the end of your current monthly payment period. To cancel go to Manage My Account or see our Digital + print Terms.By proceeding, you are agreeing to Digital + print Terms. To find out what personal data we collect and how we use it, please visit our Privacy Policy.All card payments are powered by Stripe. Read the Stripe Privacy Policy and Terms and conditions."`;
